src/com_tjvendors/admin/models/vendors.php

```diff
@@ -205,10 +205,33 @@ public function deleteVendor($vendor_id)
 	public function deleteClientFromVendor($vendor_id,$client)
 	{
 		JModelLegacy::addIncludePath(JPATH_SITE . '/components/com_tjvendors/models');
-		$tjvendorsModelVendor = JModelLegacy::getInstance('Vendor', 'TjvendorsModel');
-		$vendorData           = $tjvendorsModelVendor->getItem($vendor_id);
-		$db                   = $this->getDbo();
-		$query                = $db->getQuery(true);
+		$tjvendorsModelVendor     = JModelLegacy::getInstance('Vendor', 'TjvendorsModel');
+		$tjvendorsModelVendorFees = JModelLegacy::getInstance('VendorFees', 'TjvendorsModel');
+		$vendorData               = $tjvendorsModelVendor->getItem($vendor_id);
+		$db                       = $this->getDbo();
+
+		// Get vendor specific Fees Data
+		$tjvendorsModelVendorFees->setState('vendor_id', $vendor_id);
+		$vendorFeeData            = $tjvendorsModelVendorFees->getItems();
+
+		if (!empty($vendorFeeData))
+		{
+			foreach ($vendorFeeData as $feeData)
+			{
+				// Getting Vendor Payable Amount here
+				$result = TjvendorsHelper::getPayableAmount($feeData->vendor_id, $feeData->client, $feeData->currency);
```

Comment:
@coolbung This is the method to get the balance payable to a vendor for given client and currency. Are you saying we need method that outputs remaining payouts for all clients and all currencies?

Reply:
Yes. My suggestion is

1. Create a `getPayableAmount()` method in the vendor (singular) model with client and currency as optional inputs. Currently its in a helper which is not correct. 
2. The method always returns an array of objects and each object has the client, amount and currency.
3. If client and currency are optionally passed the array that is returned is appropriately filtered. 

This should be the one and only way to get the balance for a single vendor.

Reply:
sounds good